drivers/mspi/mspi_stm32_ospi.c

```diff
@@ -0,0 +1,1830 @@
+	int ret;
+
+	if (mspi_stm32_ospi_is_memorymap(controller)) {
+		return 0;
+	}
+	/* Configure in MemoryMapped mode */
+	if ((dev_data->dev_cfg.io_mode == MSPI_IO_MODE_SINGLE) &&
+	    (mspi_stm32_ospi_hal_address_size(dev_data->dev_cfg.addr_length) ==
+	     HAL_OSPI_ADDRESS_24_BITS)) {
+		/* OPI mode and 3-bytes address size not supported by memory */
+		LOG_ERR("MSPI_IO_MODE_SINGLE in 3Bytes addressing is not supported");
+		return -EIO;
+	}
+
+	/* Initialize the read command */
+	s_command.OperationType = HAL_OSPI_OPTYPE_READ_CFG;
+	s_command.InstructionMode = (dev_data->dev_cfg.data_rate == MSPI_DATA_RATE_SINGLE)
+					    ? ((dev_data->dev_cfg.io_mode == MSPI_IO_MODE_SINGLE)
+						       ? HAL_OSPI_INSTRUCTION_1_LINE
+						       : HAL_OSPI_INSTRUCTION_8_LINES)
+					    : HAL_OSPI_INSTRUCTION_8_LINES;
+	s_command.InstructionDtrMode = (dev_data->dev_cfg.data_rate == MSPI_DATA_RATE_SINGLE)
+					       ? HAL_OSPI_INSTRUCTION_DTR_DISABLE
+					       : HAL_OSPI_INSTRUCTION_DTR_ENABLE;
+	s_command.InstructionSize = (dev_data->dev_cfg.data_rate == MSPI_DATA_RATE_SINGLE)
+					     ? ((dev_data->dev_cfg.io_mode == MSPI_IO_MODE_SINGLE)
+							   ? HAL_OSPI_INSTRUCTION_8_BITS
+							   : HAL_OSPI_INSTRUCTION_16_BITS)
+						: HAL_OSPI_INSTRUCTION_16_BITS;
+	s_command.Instruction = (dev_data->dev_cfg.data_rate == MSPI_DATA_RATE_SINGLE)
+					? ((dev_data->dev_cfg.io_mode == MSPI_IO_MODE_SINGLE)
+						   ? ((mspi_stm32_ospi_hal_address_size(
+								   dev_data->dev_cfg.addr_length) ==
+							   HAL_OSPI_ADDRESS_24_BITS)
+								  ? SPI_NOR_CMD_READ_FAST
+								  : SPI_NOR_CMD_READ_FAST_4B)
+						   : dev_data->dev_cfg.read_cmd)
+					: SPI_NOR_OCMD_DTR_RD;
+	s_command.AddressMode = (dev_data->dev_cfg.data_rate == MSPI_DATA_RATE_SINGLE)
+					? ((dev_data->dev_cfg.io_mode == MSPI_IO_MODE_SINGLE)
+						   ? HAL_OSPI_ADDRESS_1_LINE
+						   : HAL_OSPI_ADDRESS_8_LINES)
+					: HAL_OSPI_ADDRESS_8_LINES;
+	s_command.AddressDtrMode = (dev_data->dev_cfg.data_rate == MSPI_DATA_RATE_SINGLE)
+					   ? HAL_OSPI_ADDRESS_DTR_DISABLE
+					   : HAL_OSPI_ADDRESS_DTR_ENABLE;
+	s_command.AddressSize =
+		(dev_data->dev_cfg.data_rate == MSPI_DATA_RATE_SINGLE)
+			? mspi_stm32_ospi_hal_address_size(dev_data->dev_cfg.addr_length)
+			: HAL_OSPI_ADDRESS_32_BITS;
+	s_command.DataMode = (dev_data->dev_cfg.data_rate == MSPI_DATA_RATE_SINGLE)
+					 ? ((dev_data->dev_cfg.io_mode == MSPI_IO_MODE_SINGLE)
+						? HAL_OSPI_DATA_1_LINE
+						: HAL_OSPI_DATA_8_LINES)
+					 : HAL_OSPI_DATA_8_LINES;
+	s_command.DataDtrMode = (dev_data->dev_cfg.data_rate == MSPI_DATA_RATE_SINGLE)
+					? HAL_OSPI_DATA_DTR_DISABLE
+					: HAL_OSPI_DATA_DTR_ENABLE;
+	s_command.DummyCycles = dev_data->ctx.xfer.rx_dummy;
+	s_command.DQSMode = (dev_data->dev_cfg.data_rate == MSPI_DATA_RATE_SINGLE)
+					? HAL_OSPI_DQS_DISABLE
+					: HAL_OSPI_DQS_ENABLE;
+
+	if (HAL_OSPI_Command(&dev_data->hmspi.ospi, &s_command, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) !=
+		HAL_OK) {
```

Comment:
Prefer indentation to opening parenthesis:
```suggestion
	if (HAL_OSPI_Command(&dev_data->hmspi.ospi, &s_command, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) !=
	    HAL_OK) {